.github/workflows/go_app_push_main.yml

```
@@ -44,20 +44,23 @@ jobs:
         env:
           TOKEN: ${{ secrets.GH_CI_PAT }}
           GITHUB_USERNAME: ${{ inputs.GH_CI_USER }}
-        run: git config --global url."https://${GITHUB_USERNAME}:${TOKEN}@github.com".insteadOf
+        run:
+          git config --global url."https://${GITHUB_USERNAME}:${TOKEN}@github.com".insteadOf
           "https://github.com"
       # Go vet every Go module.
       - name: go vet
-        run: find . -name vendor -prune -o -name go.mod -print | xargs -n1 dirname
+        run:
+          find . -name vendor -prune -o -name go.mod -print | xargs -n1 dirname
           | xargs -n1 -I{} bash -c "pushd {}; go vet ./..."
       # Install go-junit-report to format test results.
       - name: Install go-junit-report
         run: go install github.com/jstemmer/go-junit-report/v2@v2.1.0
       # Run unit test for evet Go module.
       - name: go test
-        run: find . -name vendor -prune -o -name go.mod -print | xargs -n1 dirname
-          | xargs -n1 -I{} bash -c "pushd {}; go test -coverprofile=coverage.txt --race -v ./..."
-          | go-junit-report -set-exit-code > junit_report.xml || true
+        run: |
+          find . -name vendor -prune -o -name go.mod -print | xargs -n1 dirname \
+          | xargs -n1 -I{} bash -c "pushd {}; go test -coverprofile=coverage.txt --race -v ./..." \
+          | tee >(go-junit-report -set-exit-code > junit_report.xml)
```

Maybe I'm over simplifying this, but instead of piping into junit at all, why not just run it on it's own line to get all the outputs?
```suggestion
        run: |
          find . -name vendor -prune -o -name go.mod -print | xargs -n1 dirname \
          | xargs -n1 -I{} bash -c "pushd {}; go test -coverprofile=coverage.txt --race -v ./..."
          go-junit-report -set-exit-code > junit_report.xml
```